### ISSUE TITLE
Error `'a' (<dimension1> x 1) must be square`

### ISSUE THREAD

**Issue by fweber144** [melff/mclogit#22]
For some models fitted by `mclogit::mblogit()`, I get the error
```
Error in solve.default(X[[i]], ...) : 'a' (<dimension1> x 1) must be square
```
thrown by `mclogit:::PQLMQL_innerFit()`. (The `<dimension1>` can be `3`, for example.)

I think the problematic line is https://github.com/melff/mclogit/blob/7f47d53713b124a100538bfd65dc899c320ffa4f/pkg/R/mmclogit-fitPQLMQL.R#L271 because, if `S.k` is a matrix there, then `diag()` returns a *vector* containing the diagonal elements. Perhaps that line was supposed to read
```r
S.k <- diag(x=diag(S.k),nrow=d)
```
instead? But I don't know anything about the mathematical details of this line, so my guess might be completely wrong.

And I'm not sure, but perhaps this is the same bug as in #21?

Note that this issue occurs with both, the current CRAN version 0.8.7.3 as well as with the current GitHub version at commit 7f47d53713b124a100538bfd65dc899c320ffa4f installed via `devtools::install_github("melff/mclogit",subdir="pkg")`.

**Comment by Zhiwen-Owen-Jiang** [comment]
I found this bug either. The code that previously I could run failed.

**Comment by melff** [comment]
This appears to be fixed in release 0.9.4.

**Comment by fweber144** [comment]
Thanks, this seems indeed to be fixed. However, I now get an error from `predict.mmblogit()` which I'll open a new issue for.
